Libraries/JavaScriptAppEngine/System/JSTimers/JSTimersExecution.js

```diff
@@ -70,6 +73,12 @@ var JSTimersExecution = {
       } else if (type === JSTimersExecution.Type.requestAnimationFrame) {
         var currentTime = performanceNow();
         callback(currentTime);
+      } else if (type === JSTimersExecution.Type.requestIdleCallback) {
+        callback({
+          timeRemaining: function() {
+            return Math.max(0, 17 - (performanceNow() - frameTime));
```

Comment:
Yeah, the way we'd want to do this at first is to check if there are any pending JS calls to process. We can/do keep track of this in the native bridge, but this isn't exposed right now.

As far as debugging, the issue with the sync hook is less debugging this in particular and more being able to debug an application that uses this API at all: an app that uses this API cannot run in Chrome.

Internally, we have a lot of reasons for wanting an on-device debugger but are still looking for someone to take the project on.